### ISSUE TITLE
Password for accessing "private" photos

### ISSUE THREAD

**Issue by hialvaro** [photoprism/photoprism#325]
I think it would be nice to be able to set a password to access the "Private" photos.

**Comment by lastzero** [comment]
You are absolutely right, that makes sense. What you see now is just a first step to separate "potentially problematic" photos (like when you were drunk and/or naked) from the others, so that they don't randomly show up in the wrong moment.

**Comment by uxusoft** [comment]
Hi! Is it possible to add golden label for this feature request.

**Comment by graciousgrey** [comment]
@uxusoft sure I just added your label :)

**Comment by XtremeOwnageDotCom** [comment]
To note, I attempted to lock the page behind some basic authentication using my proxy. 

Since- the application is a SPA using VUE, this will only prevent access to the "private" area if you attempt to navigate directly to it from an external URL. With that said, this means the page cannot be blocked externally via a proxy. So.... yea, it would be quite nice to keep this as a publicly accessible service on my local network, for ya know, sharing family photos and stuff........... without some of the um... "NSFW" photos being clickable.

Because, when somebody sees the navigation options on the left, the first thing they are going to want to do, is to click the "PRIVATE" button. Because, that sounds interesting.

As well, since the application does not have support for individual user accounts, it is not currently possible to give access to this application WITHOUT the giant "private" button appearing in the navigation menu.

So... yea, pretty important feature request here.

**Comment by graciousgrey** [comment]
At the moment you shouldn't give someone access to your user account if they should not be able to see your private photos.

We are currently working on multi-user support. This will allow you to create a read-only users who won't have access to private photos (neither using the url nor in any search results).

**Comment by klibansky** [comment]
Really like this feature. It would be quite unique because I haven't found this feature in any other photo album app I have found.
Don't forget to remove (or blur) the thumbnail/cache images.

Another feature I would love to have, connected to this one, is an extra layer of security, by encrypting the source files somehow. I know there will be a performance hit but better safe than sorry.
I would be kind of like having all the NSFW pics in a password protected zip file in the background.

**Comment by gyto6** [comment]
> Another feature I would love to have, connected to this one, is an extra layer of security, by encrypting the source files somehow. I know there will be a performance hit but better safe than sorry. I would be kind of like having all the NSFW pics in a password protected zip file in the background.

First, encryption is a thing that must be done by your filesystem. Especially to not suffer from performances issues. Multi-User being on its way, you can't cipher all users photos with one key. It'd wouldn't be logical and secure. BTW, it would require for "guest" user to remind about a key that encrypt their whole pictures, and you can be sure that they'll loose it. 

Second, zip all NSFW pictures is not handy for photo gallery. Secured by a password btw, it does less sense. Photoprism is a website Photo Gallery, so access are governed by the SQL schema. Users can't access to data as they do not access the filesystem. 

Set encryption on your filesystem, and use Nextcloud. Or use NextCloud per user encryption to give any of them a special cipher key for encryption. Contrary to NextCloud, Photoprism is a Photo Gallery / Manager (?), but not a file manager or "drive" solution. It's not supposed to edit your pictures by now, exception should be to add metadata to complete photos informations.

**Comment by XtremeOwnageDotCom** [comment]
Y'all quit trying to add "feature creep" here. Let's get to a point where the NSFW/Private photos can at least have a password and then move on!

**Comment by JAKAMI99** [comment]
One year later. Still wanted :)

**Comment by lastzero** [comment]
We know, but have been releasing a long list of other features and enhancements. So it's not like we did nothing. Please let us enjoy our vacation. We'll work on this and all the other open feature requests. Thank you very much! ❤️

**Comment by JAKAMI99** [comment]
> We know, but have been releasing a long list of other features and enhancements. So it's not like we did nothing. Please let us enjoy our vacation. We'll work on this and all the other open feature requests. Thank you very much! ❤️

No need for excuses :(
I think my comment got misunderstood.
Just wanted to tell that somebody else (me) would like this feature...
I've never told, you did nothing :/ of course I let you enjoy your vacation...
That escalated a bit :/
*virtual hug*

**Comment by lastzero** [comment]
No worries. Simply give this issue a thumbs up at the bottom of the description in this case. We had many users asking for status updates in the recent days, which triggers an email notification to all contributors every time. We therefore recommend to use GitHub Discussions for questions and comments.

**Comment by ahlongxp** [comment]
I need this feature!!

**Comment by Akxe** [comment]
Maybe then the private photos can be searchable if you "log in" for the photos... I could imagine every user in a family having his own login for the private section...

**Comment by fl0wm0ti0n** [comment]
hello, right now the private function is just useless (in my opinion)...
well they wont show up maybe in some sections but everyone can just click on private and see them...

so please just a password! cant be that hard :D thanks!

**Comment by Arslan-Rja** [comment]
I would also like to express interest in this feature.

**Comment by avsamano** [comment]
Jumping in to also request this feature and explain my use-case:

I'm not interested in this because of login sharing.
Rather, I'm interested in this for when I'm simply showing pictures to someone in-person.
Considering the potentially sensitive nature of the images, an errant click by myself or the person I'm showing pictures to could be very embarrassing.

Having that extra layer of a password on Private pictures would just provide piece of mind when scrolling through pictures with someone.

Thank you for the awesome software.

**Comment by Ranger21** [comment]
That would be absolutely great idea!

**Comment by superuser866** [comment]
I also think it would be great just asking for a numeric PIN when clicking the Private button in order to avoid opening the Private section by mistake when showing pictures or navigating the menu in presence of other people. 
Thank you very much!

**Comment by nirajbhatt7** [comment]
Let me upvote this in 2026.
